packages/order/src/models/order.ts

```diff
@@ -0,0 +1,168 @@
+import { DAL } from "@medusajs/types"
+import {
+  OrderStatus,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Cascade,
+  Collection,
+  Entity,
+  Enum,
+  ManyToOne,
+  OnInit,
+  OneToMany,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import Address from "./address"
+import LineItem from "./line-item"
+import ShippingMethod from "./shipping-method"
+
+type OptionalOrderProps =
+  | "shipping_address"
+  | "billing_address"
+  | DAL.EntityDateColumns
+
+const regionIdIndex = createPsqlIndexStatementHelper({
+  tableName: "order",
+  columns: "region_id",
+  where: "deleted_at IS NOT NULL",
+})
+
+const customerIdIndex = createPsqlIndexStatementHelper({
+  tableName: "order",
+  columns: "customer_id",
+  where: "deleted_at IS NOT NULL",
+})
+
+const salesChannelIdIndex = createPsqlIndexStatementHelper({
+  tableName: "order",
+  columns: "customer_id",
+  where: "deleted_at IS NOT NULL",
+})
+
+const orderDeletedAtIndex = createPsqlIndexStatementHelper({
+  tableName: "order",
+  columns: "deleted_at",
+  where: "deleted_at IS NOT NULL",
+})
+
+const currencyCodeIndex = createPsqlIndexStatementHelper({
+  tableName: "order",
+  columns: "currency_code",
+  where: "deleted_at IS NOT NULL",
+})
+
+@Entity({ tableName: "order" })
+export default class Order {
```

Comment:
todo: add a canceled_at field here.

Reply:
We might also want to have a `placed_at` here. E.g., to indicate when an order moves from a draft state to being placed/confirmed.